gokart/tree/task_info.py

```diff
@@ -78,3 +79,30 @@ def dump_task_info_table(task: TaskOnKart, task_info_dump_path: str, ignore_task
 
     task_info_target = make_target(file_path=task_info_dump_path, unique_id=unique_id)
     task_info_target.dump(obj=task_info_table, lock_at_dump=False)
+
+
+def dump_task_info_tree(task: TaskOnKart, task_info_dump_path: str, ignore_task_names: Optional[List[str]]):
+    """Dump the task info tree object (TaskInfo) to a pickle file.
+
+    Parameters
+    ----------
+    - task: TaskOnKart
+        Root task.
+    - task_info_dump_path: str
+        Output target file path. Path destination can be `local`, `S3`, or `GCS`.
+        File extension must be '.pkl'.
+    - ignore_task_names: Optional[List[str]]
+        List of task names to ignore.
+    Returns
+    -------
+    None
+    """
+    extension = os.path.splitext(task_info_dump_path)[1]
+    assert extension == '.pkl', f'File extention must be `.pkl`, not `{extension}`.'
+
+    task_info_tree = make_task_info_tree(task, ignore_task_names=ignore_task_names)
+
+    unique_id = task.make_unique_id()
```

Comment:
@mski-iksm [nits]
I think it's handy to add use_unique_id option and following statement.

```
        unique_id = self.make_unique_id() if use_unique_id else None
``

Reply:
@Hi-king I've add use_unique_id parameter. Please check!

Reply:
@mski-iksm Great! Thanks.